src/util.cpp

```diff
@@ -592,26 +592,41 @@ fs::path GetConfigFile(const std::string& confPath)
     return pathConfigFile;
 }
 
+void ArgsManager::ReadConfigStream(fs::ifstream& streamConfig)
+{
+    std::set<std::string> setOptions;
+    setOptions.insert("*");
+
+    for (boost::program_options::detail::config_file_iterator it(streamConfig, setOptions), end; it != end; ++it) {
+        // Don't overwrite existing settings so command line settings override bitcoin.conf
+        std::string strKey = std::string("-") + it->string_key;
+        std::string strValue = it->value[0];
+        InterpretNegativeSetting(strKey, strValue);
+        if (mapArgs.count(strKey) == 0) {
+            mapArgs[strKey] = strValue;
+        }
+        mapMultiArgs[strKey].push_back(strValue);
+    }
+}
+
 void ArgsManager::ReadConfigFile(const std::string& confPath)
 {
     fs::ifstream streamConfig(GetConfigFile(confPath));
     if (!streamConfig.good())
         return; // No bitcoin.conf file is OK
 
+    std::string includeconf;
     {
         LOCK(cs_args);
-        std::set<std::string> setOptions;
-        setOptions.insert("*");
-
-        for (boost::program_options::detail::config_file_iterator it(streamConfig, setOptions), end; it != end; ++it)
-        {
-            // Don't overwrite existing settings so command line settings override bitcoin.conf
-            std::string strKey = std::string("-") + it->string_key;
-            std::string strValue = it->value[0];
-            InterpretNegativeSetting(strKey, strValue);
-            if (mapArgs.count(strKey) == 0)
-                mapArgs[strKey] = strValue;
-            mapMultiArgs[strKey].push_back(strValue);
+        ReadConfigStream(streamConfig);
+        if (mapArgs.count("-includeconf")) includeconf = mapArgs["-includeconf"];
+    }
+    if (includeconf != "") {
+        fs::path includeFile = GetConfigFile(includeconf);
+        fs::ifstream includeConfig(includeFile);
+        if (includeConfig.good()) {
```

Comment:
Can we have an else clause that prints an error message if we fail to open an includeconfig file?